lib/include/tick/base_model/model.h

```diff
@@ -10,12 +10,16 @@
 #include "tick/base/base.h"
 
 #include <cereal/cereal.hpp>
+#include <cereal/types/polymorphic.hpp>
 
 #include <iostream>
 
 // TODO: Model "data" : ModeLabelsFeatures, Model,Model pour les Hawkes
 template <class T>
 class TModel {
+  template <class T1>
+  friend std::ostream &operator<<(std::ostream &, const TModel<T1> &);
```

Comment:
Why is T1 needed, can't it be T?

Reply:
T1 is required as it's a thing called "template shadowing" 

it's good in this case to let the compiler know that May be a different type(not that will be)